src/pluggy/manager.py

```diff
@@ -325,11 +326,24 @@ def add_hookcall_monitoring(self, before, after):
         same arguments as ``before`` but also a :py:class:`_Result`` object
         which represents the result of the overall hook call.
         """
-        return _tracing._TracedHookExecution(self, before, after).undo
+        oldcall = self._inner_hookexec
```

Comment:
I think it is useful to keep the previous assert:

```
assert not isinstance(self.oldcall, _TracedHookExecution)
```

Reply:
Just checking, did you see the commit message and disagree?

Reply:
No I didn't! Thanks for the pointer, I appreciate the well written commit messages.

I agree with the reasoning there, so dismiss this comment.